### PR TITLE
fix(shadow): Fix shadow support in locator.js

### PR DESCRIPTION
**Pull request by patrjam**
## Motivation/Description of the PR
- We need to use scrollTo method for scrolling on shadow elements - it didnt work correctly for this case.
- We also use resemble helper for visual testing, which uses webdriver methods and here was also problem with shadow selectors. This PR resolves both.
- Resolves https://github.com/codeceptjs/CodeceptJS/issues/2585 

Applicable helpers:

- [ ] WebDriver

## Type of change

- [x] :bug: Bug fix

## Checklist:

- [ok] Tests have been added
- [ok] Documentation has been added (Run `npm run docs`)
- [ok] Lint checking (Run `npm run lint`)
- [ok] Local tests are passed (Run `npm test`)

### CODE REVIEW

File: test/unit/locator_test.js
Comment (DavertMik) [inline test/unit/locator_test.js:89]: JavaScript object written as a string? 
A very strange design decision... :confused:

**Comment by patrjam** [comment]
Still waiting for review. We need it ASAP.

@DavertMik

File: test/unit/locator_test.js
Comment (DavertMik) [inline test/unit/locator_test.js:89]: sorry, I still don't understand why do you pass it as string if you can pass it by array:

```js
{ shadow: ["my-app", "recipe-hello-binding", "ui-input", "input.input"] }
```

File: lib/locator.js
Comment (DavertMik) [inline lib/locator.js:356]: no, this is a bad way to detect shadow locator

**Review by DavertMik — CHANGES_REQUESTED** [review]
I understand the issue but I don't agree on the way you fix it. 
Sure, we need shadow locators but we need to have a better policy on that.

Shadow locator must always be: object with array as parameters.

```js
{ shadow: ['a', 'b', 'c'] }
```

so isShadow should detect that value is object with `shadow` key.

**Comment by patrjam** [comment]
Updated.

**Comment by patrjam** [comment]
Still waiting for review. @DavertMik 
Thanks.

**Comment by patrjam** [comment]
Should review anyone else, please? Iam still waiting... e.g. @Vorobeyko, @PeterNgTr

**Comment by Trawais** [comment]
Please guys. We can't wait for this one being fixed. @DavertMik @Vorobeyko @PeterNgTr @Georgegriff @elukoyanov Anybody else?

**Comment by Georgegriff** [comment]
This pr is targeting the wrong branch please rebase to 3.x

**Comment by DavertMik** [comment]
Thanks, all checks passed, should be fine to merge. Thank you. This fix will go to 3.0.3-beta

**Pull request merged**
